docs/proposals/2170-kubeflow-training-v2/README.md

```diff
@@ -301,29 +301,42 @@ type TrainJobSpec struct {
 	Suspend *bool `json:"suspend,omitempty"`
 
 	// ManagedBy is used to indicate the controller or entity that manages a TrainJob.
+	// The value must be either an empty, 'training-operator.kubeflow.org/trainjob-controller' or
+	// 'kueue.x-k8s.io/multikueue'.
+	// The built-in TrainJob controller reconciles TrainJob which don't have this
+	// field at all or the field value is the reserved string
+	// 'training-operator.kubeflow.org/trainjob-controller', but delegates reconciling TrainJobs
+	// with a 'kueue.x-k8s.io/multikueue' to the Kueue.
+	//
+	// The value must be a valid domain-prefixed path (e.g. acme.io/foo) -
+	// all characters before the first "/" must be a valid subdomain as defined
+	// by RFC 1123. All characters trailing the first "/" must be valid HTTP Path
+	// characters as defined by RFC 3986. The value cannot exceed 63 characters.
+	// The field is immutable.
 	ManagedBy *string `json:"managedBy,omitempty"`
 }
 
 type TrainingRuntimeRef struct {
-	// Name for the training runtime.
+	// Name for the runtime.
+	// This must indicate the runtime deployed in the same namespace as the TrainJob
+	// when TrainingRuntime is used in the kind.
 	Name string `json:"name"`
 
-	// Namespace for the runtime. In that case, user should use TrainingRuntime
-	Namespace *string `json:"namespace,omitempty"`
+	// APIVersion is the apiVersion for the runtime.
+	// Defaults to the v2alpha1.
+	APIVersion *string `json:apiVersion,omitempty`
 
-	// Kind for the runtime. TrainingRuntime or ClusterTrainingRuntime
+	// Kind for the runtime, which must be TrainingRuntime or ClusterTrainingRuntime.
+	// Defaults to the TrainingRuntime.
 	Kind *string `json:"kind,omitempty"`
-
-	// Version for the runtime. For example: v2alpha1
-	Version *string `json:"version,omitempty"`
```

Comment:
Are we going to introduce Revision parameter in the future versions according to this ?
https://github.com/kubeflow/training-operator/blob/82d6fe581bacb0b4961b527f00385ece8fc1311c/docs/proposals/2170-kubeflow-training-v2/README.md#:~:text=In%20the%20future%20implementation%2C%20we%20will%20introduce%20a%20revision%20control%20mechanism%20similar%20to%20Kubernetes%20Deployment%20to%20control%20versions%20of%20TrainingRuntime%20and%20enable%20rolling%20updates.

Reply:
Yeah, I still think that the runtime revision control would be useful.